File: scala/SimpleAk/src/main/scala-2.11/actor/PongActor.scala
package actor

import akka.actor.{Props, Status, Actor}

import scala.util.control.Exception

/**
 * Created by cqshinn on 2/12/16.
 */

object PongActor{
  implicit val args: String = "Default"
  def props(implicit response: String): Props ={
    return Props(classOf[PongActor],response)
  }
}
class PongActor(implicit name: String) extends Actor{
  override def receive = {
    case "Pong" => {
      println(s"Ping $name")
      sender ! s"Ping $name"
    }
    case _ => sender ! Status.Failure(new Exception("unknown message"))
  }
}

File: java/SimpleAk/src/main/java/actor/PongActor.java
package actor;

import akka.actor.ActorRef;
import akka.actor.Props;
import akka.actor.Status;
import akka.actor.UntypedActor;

/**
 * Created by cqshinn on 2/12/16.
 */
public class PongActor extends UntypedActor {
    @Override
    public void onReceive(Object message) throws Exception {
        if(message instanceof String){
            if(message.equals("Ping")){
                sender().tell("Pong", ActorRef.noSender());
            }
        }
        else {
            sender().tell(new Status.Failure(new Exception("unknown message")), self());
        }
    }

    public static Props props(String args){
        return Props.create(PongActor.class,args);
    }
}
